pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -4,10 +4,14 @@ addCMakeParams() {
 
 fixCmakeFiles() {
     # Replace occurences of /usr and /opt by /var/empty.
+    # However, avoid replacing when /usr appears in the middle of a path with
+    # _CMAKE_OSX_SYSROOT_PATH or sdk_path variables preceding it. Those cases
+    # happen in the files related to the Darwin platform, and they relate to
+    # the Xcode SDK. Replacing by /var/empty in those cases would be incorrect.
     echo "fixing cmake files..."
     find "$1" \( -type f -name "*.cmake" -o -name "*.cmake.in" -o -name CMakeLists.txt \) -print |
         while read fn; do
-            sed -e 's^/usr\([ /]\|$\)^/var/empty\1^g' -e 's^/opt\([ /]\|$\)^/var/empty\1^g' < "$fn" > "$fn.tmp"
+            sed -e '/\${\(_CMAKE_OSX_SYSROOT_PATH\|sdk_path\)}\/usr/! s^/usr\([ /]\|$\)^/var/empty\1^g' -e 's^/opt\([ /]\|$\)^/var/empty\1^g' < "$fn" > "$fn.tmp"
```

Comment:
```suggestion
            sed -e 's#\(\s\|"\|^\)/usr\(\s\|/\|$\)#\1/var/empty\2#g' -e 's#\(\s\|"\|^\)/opt\(\s\|/\|$\)#\1/var/empty\2#g' < "$fn" > "$fn.tmp"
```

Reply:
Diff before and after:
```diff
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCUDACompiler.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCUDACompiler.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCUDACompiler.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCUDACompiler.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -179,19 +179,19 @@
     # CMAKE_CUDA_COMPILER_LIBRARY_ROOT contains the device library.
     if(EXISTS "${CMAKE_CUDA_COMPILER_TOOLKIT_ROOT}/nvvm/libdevice")
       set(CMAKE_CUDA_COMPILER_LIBRARY_ROOT "${CMAKE_CUDA_COMPILER_TOOLKIT_ROOT}")
-    elseif(CMAKE_SYSROOT_LINK AND EXISTS "${CMAKE_SYSROOT_LINK}/var/empty/lib/cuda/nvvm/libdevice")
-      set(CMAKE_CUDA_COMPILER_LIBRARY_ROOT "${CMAKE_SYSROOT_LINK}/var/empty/lib/cuda")
-    elseif(EXISTS "${CMAKE_SYSROOT}/var/empty/lib/cuda/nvvm/libdevice")
-      set(CMAKE_CUDA_COMPILER_LIBRARY_ROOT "${CMAKE_SYSROOT}/var/empty/lib/cuda")
+    elseif(CMAKE_SYSROOT_LINK AND EXISTS "${CMAKE_SYSROOT_LINK}/usr/lib/cuda/nvvm/libdevice")
+      set(CMAKE_CUDA_COMPILER_LIBRARY_ROOT "${CMAKE_SYSROOT_LINK}/usr/lib/cuda")
+    elseif(EXISTS "${CMAKE_SYSROOT}/usr/lib/cuda/nvvm/libdevice")
+      set(CMAKE_CUDA_COMPILER_LIBRARY_ROOT "${CMAKE_SYSROOT}/usr/lib/cuda")
     else()
       message(FATAL_ERROR "Couldn't find CUDA library root.")
     endif()
 
     # CMAKE_CUDA_COMPILER_TOOLKIT_LIBRARY_ROOT contains the linking stubs necessary for device linking and other low-level library files.
-    if(CMAKE_SYSROOT_LINK AND EXISTS "${CMAKE_SYSROOT_LINK}/var/empty/lib/nvidia-cuda-toolkit/bin/crt/link.stub")
-      set(CMAKE_CUDA_COMPILER_TOOLKIT_LIBRARY_ROOT "${CMAKE_SYSROOT_LINK}/var/empty/lib/nvidia-cuda-toolkit")
-    elseif(EXISTS "${CMAKE_SYSROOT}/var/empty/lib/nvidia-cuda-toolkit/bin/crt/link.stub")
-      set(CMAKE_CUDA_COMPILER_TOOLKIT_LIBRARY_ROOT "${CMAKE_SYSROOT}/var/empty/lib/nvidia-cuda-toolkit")
+    if(CMAKE_SYSROOT_LINK AND EXISTS "${CMAKE_SYSROOT_LINK}/usr/lib/nvidia-cuda-toolkit/bin/crt/link.stub")
+      set(CMAKE_CUDA_COMPILER_TOOLKIT_LIBRARY_ROOT "${CMAKE_SYSROOT_LINK}/usr/lib/nvidia-cuda-toolkit")
+    elseif(EXISTS "${CMAKE_SYSROOT}/usr/lib/nvidia-cuda-toolkit/bin/crt/link.stub")
+      set(CMAKE_CUDA_COMPILER_TOOLKIT_LIBRARY_ROOT "${CMAKE_SYSROOT}/usr/lib/nvidia-cuda-toolkit")
     else()
       set(CMAKE_CUDA_COMPILER_TOOLKIT_LIBRARY_ROOT "${CMAKE_CUDA_COMPILER_TOOLKIT_ROOT}")
     endif()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCompiler.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCompiler.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCompiler.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/CMakeDetermineCompiler.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -87,7 +87,7 @@
     endmacro()
 
     set(xcrun_result)
-    if (CMAKE_${lang}_COMPILER MATCHES "^/var/empty/bin/(.+)$")
+    if (CMAKE_${lang}_COMPILER MATCHES "^/usr/bin/(.+)$")
       _query_xcrun("${CMAKE_MATCH_1}" RESULT_VAR xcrun_result)
     elseif (CMAKE_${lang}_COMPILER STREQUAL "CMAKE_${lang}_COMPILER-NOTFOUND")
       foreach(comp ${CMAKE_${lang}_COMPILER_LIST})
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDA.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDA.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDA.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDA.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -39,8 +39,8 @@
 path and ``REQUIRED`` is specified to :command:`find_package`.  To use
 a different installed version of the toolkit set the environment variable
 ``CUDA_BIN_PATH`` before running cmake (e.g.
-``CUDA_BIN_PATH=/var/empty/local/cuda1.0`` instead of the default
-``/var/empty/local/cuda``) or set ``CUDA_TOOLKIT_ROOT_DIR`` after configuring.  If
+``CUDA_BIN_PATH=/usr/local/cuda1.0`` instead of the default
+``/usr/local/cuda``) or set ``CUDA_TOOLKIT_ROOT_DIR`` after configuring.  If
 you change the value of ``CUDA_TOOLKIT_ROOT_DIR``, various components that
 depend on the path will be relocated.
 
@@ -943,7 +943,7 @@
   if(APPLE)
     # We need to add the default path to the driver (libcuda.dylib) as an rpath, so that
     # the static cuda runtime can find it at runtime.
-    list(APPEND CUDA_LIBRARIES -Wl,-rpath,/var/empty/local/cuda/lib)
+    list(APPEND CUDA_LIBRARIES -Wl,-rpath,/usr/local/cuda/lib)
   endif()
 else()
   list(APPEND CUDA_LIBRARIES ${CUDA_CUDART_LIBRARY})
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDAToolkit.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDAToolkit.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDAToolkit.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindCUDAToolkit.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -37,7 +37,7 @@
    responsible for ensuring that the first ``nvcc`` to show up in the path is
    the desired path in the event that multiple CUDA Toolkits are installed.
 
-5. On Unix systems, if the symbolic link ``/var/empty/local/cuda`` exists, this is
+5. On Unix systems, if the symbolic link ``/usr/local/cuda`` exists, this is
    used.  No subsequent search attempts are performed.  No default symbolic link
    location exists for the Windows platform.
 
@@ -50,20 +50,20 @@
    +=============+=============================================================+
    | macOS       | ``/Developer/NVIDIA/CUDA-X.Y``                              |
    +-------------+-------------------------------------------------------------+
-   | Other Unix  | ``/var/empty/local/cuda-X.Y``                                     |
+   | Other Unix  | ``/usr/local/cuda-X.Y``                                     |
    +-------------+-------------------------------------------------------------+
    | Windows     | ``C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\vX.Y`` |
    +-------------+-------------------------------------------------------------+
 
    Where ``X.Y`` would be a specific version of the CUDA Toolkit, such as
-   ``/var/empty/local/cuda-9.0`` or
+   ``/usr/local/cuda-9.0`` or
    ``C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0``
 
    .. note::
 
        When multiple CUDA Toolkits are installed in the default location of a
-       system (e.g., both ``/var/empty/local/cuda-9.0`` and ``/var/empty/local/cuda-10.0``
-       exist but the ``/var/empty/local/cuda`` symbolic link does **not** exist), this
+       system (e.g., both ``/usr/local/cuda-9.0`` and ``/usr/local/cuda-10.0``
+       exist but the ``/usr/local/cuda`` symbolic link does **not** exist), this
        package is marked as **not** found.
 
        There are too many factors involved in making an automatic decision in
@@ -540,10 +540,10 @@
       set(${result_variable} "${CUDAToolkit_ROOT}/version.txt" PARENT_SCOPE)
     elseif(CUDAToolkit_ROOT_DIR AND EXISTS "${CUDAToolkit_ROOT_DIR}/version.txt")
       set(${result_variable} "${CUDAToolkit_ROOT_DIR}/version.txt" PARENT_SCOPE)
-    elseif(CMAKE_SYSROOT_LINK AND EXISTS "${CMAKE_SYSROOT_LINK}/var/empty/lib/cuda/version.txt")
-      set(${result_variable} "${CMAKE_SYSROOT_LINK}/var/empty/lib/cuda/version.txt" PARENT_SCOPE)
-    elseif(EXISTS "${CMAKE_SYSROOT}/var/empty/lib/cuda/version.txt")
-      set(${result_variable} "${CMAKE_SYSROOT}/var/empty/lib/cuda/version.txt" PARENT_SCOPE)
+    elseif(CMAKE_SYSROOT_LINK AND EXISTS "${CMAKE_SYSROOT_LINK}/usr/lib/cuda/version.txt")
+      set(${result_variable} "${CMAKE_SYSROOT_LINK}/usr/lib/cuda/version.txt" PARENT_SCOPE)
+    elseif(EXISTS "${CMAKE_SYSROOT}/usr/lib/cuda/version.txt")
+      set(${result_variable} "${CMAKE_SYSROOT}/usr/lib/cuda/version.txt" PARENT_SCOPE)
     endif()
   endfunction()
 
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindGSL.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindGSL.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindGSL.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindGSL.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -91,7 +91,7 @@
 # locations.  These find_library and find_path calls will prefer custom
 # locations over standard locations (HINTS).  If the requested file is not found
 # at the HINTS location, standard system locations will be still be searched
-# (/var/empty/lib64 (Redhat), lib/i386-linux-gnu (Debian)).
+# (/usr/lib64 (Redhat), lib/i386-linux-gnu (Debian)).
 
 find_path( GSL_INCLUDE_DIR
   NAMES gsl/gsl_sf.h
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindMPI.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindMPI.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindMPI.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindMPI.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -610,7 +610,7 @@
 
   # For GNU-style compilers, it's possible to prefix includes and definitions with certain flags to pass them
   # only to the preprocessor. For CMake purposes, we need to treat, but ignore such scopings.
-  # Note that we do not support spaces between the arguments, i.e. -Wp,-I -Wp,/var/empty/mympi will not be parsed
+  # Note that we do not support spaces between the arguments, i.e. -Wp,-I -Wp,/opt/mympi will not be parsed
   # correctly. This form does not seem to occur in any common MPI implementation, however.
   if(NOT MSVC)
     set(_MPI_PREPROCESSOR_FLAG_REGEX "(-Wp,|-Xpreprocessor )?")
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindPatch.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindPatch.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/FindPatch.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/FindPatch.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -35,9 +35,9 @@
 if(CMAKE_HOST_WIN32)
   set(_patch_path
     "$ENV{LOCALAPPDATA}/Programs/Git/bin"
-    "$ENV{LOCALAPPDATA}/Programs/Git/var/empty/bin"
+    "$ENV{LOCALAPPDATA}/Programs/Git/usr/bin"
     "$ENV{APPDATA}/Programs/Git/bin"
-    "$ENV{APPDATA}/Programs/Git/var/empty/bin"
+    "$ENV{APPDATA}/Programs/Git/usr/bin"
     )
 endif()
 
@@ -53,7 +53,7 @@
   # prefixes on Windows.
   find_program(Patch_EXECUTABLE
     NAMES patch
-    PATH_SUFFIXES Git/var/empty/bin Git/bin GnuWin32/bin
+    PATH_SUFFIXES Git/usr/bin Git/bin GnuWin32/bin
     DOC ${_doc}
     )
 endif()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/GNUInstallDirs.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/GNUInstallDirs.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/GNUInstallDirs.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/GNUInstallDirs.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -53,7 +53,7 @@
 ``INCLUDEDIR``
   C header files (``include``)
 ``OLDINCLUDEDIR``
-  C header files for non-gcc (``/var/empty/include``)
+  C header files for non-gcc (``/usr/include``)
 ``DATAROOTDIR``
   read-only architecture-independent data root (``share``)
 ``DATADIR``
@@ -95,13 +95,13 @@
   For example, the ``SYSCONFDIR`` value ``etc`` becomes ``/etc``.
   This is required by the `GNU Coding Standards`_.
 
-``/var/empty/...``
+``/opt/...``
 
   For ``<dir>`` equal to ``SYSCONFDIR``, ``LOCALSTATEDIR`` or
   ``RUNSTATEDIR``, the ``CMAKE_INSTALL_FULL_<dir>`` is computed by
   *appending* the prefix to the value of ``CMAKE_INSTALL_<dir>``
   if it is not user-specified as an absolute path.
-  For example, the ``SYSCONFDIR`` value ``etc`` becomes ``/etc/var/empty/...``.
+  For example, the ``SYSCONFDIR`` value ``etc`` becomes ``/etc/opt/...``.
   This is defined by the `Filesystem Hierarchy Standard`_.
 
 .. _`Filesystem Hierarchy Standard`: https://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html
@@ -229,11 +229,11 @@
       AND NOT EXISTS "/etc/arch-release")
     if (EXISTS "/etc/debian_version") # is this a debian system ?
       if(CMAKE_LIBRARY_ARCHITECTURE)
-        if("${CMAKE_INSTALL_PREFIX}" MATCHES "^/var/empty/?$")
+        if("${CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
           set(_LIBDIR_DEFAULT "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
         endif()
         if(DEFINED _GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX
-            AND "${_GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX}" MATCHES "^/var/empty/?$")
+            AND "${_GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
           set(__LAST_LIBDIR_DEFAULT "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
         endif()
       endif()
@@ -264,7 +264,7 @@
     AND NOT CMAKE_CROSSCOMPILING
     AND NOT EXISTS "/etc/arch-release"
     AND EXISTS "/etc/debian_version" # is this a debian system ?
-    AND "${CMAKE_INSTALL_PREFIX}" MATCHES "^/var/empty/?$")
+    AND "${CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
   # see https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html#usrlibexec
   # and https://www.debian.org/doc/debian-policy/ch-opersys#file-system-structure (section 9.1.1 bullet point 4)
   _GNUInstallDirs_cache_path(CMAKE_INSTALL_LIBEXECDIR "${CMAKE_INSTALL_LIBDIR}"
@@ -276,7 +276,7 @@
 _GNUInstallDirs_cache_path(CMAKE_INSTALL_INCLUDEDIR "include"
   "C header files (include)")
 _GNUInstallDirs_cache_path(CMAKE_INSTALL_OLDINCLUDEDIR "/var/empty/include"
-  "C header files for non-gcc (/var/empty/include)")
+  "C header files for non-gcc (/usr/include)")
 _GNUInstallDirs_cache_path(CMAKE_INSTALL_DATAROOTDIR "share"
   "Read-only architecture-independent data root (share)")
 
@@ -348,13 +348,13 @@
         endif()
         set(${absvar} "/${${var}}")
       endif()
-    elseif("${CMAKE_INSTALL_PREFIX}" MATCHES "^/var/empty/?$")
+    elseif("${CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
       if("${dir}" STREQUAL "SYSCONFDIR" OR "${dir}" STREQUAL "LOCALSTATEDIR" OR "${dir}" STREQUAL "RUNSTATEDIR")
         set(${absvar} "/${${var}}")
       else()
         set(${absvar} "${CMAKE_INSTALL_PREFIX}/${${var}}")
       endif()
-    elseif("${CMAKE_INSTALL_PREFIX}" MATCHES "^/var/empty/.*")
+    elseif("${CMAKE_INSTALL_PREFIX}" MATCHES "^/opt/.*")
       if("${dir}" STREQUAL "SYSCONFDIR" OR "${dir}" STREQUAL "LOCALSTATEDIR" OR "${dir}" STREQUAL "RUNSTATEDIR")
         set(${absvar} "/${${var}}${CMAKE_INSTALL_PREFIX}")
       else()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/GetPrerequisites.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/GetPrerequisites.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/GetPrerequisites.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/GetPrerequisites.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -513,13 +513,13 @@
     string(TOLOWER "${resolved_file}" lower)
 
     if(UNIX)
-      if(resolved_file MATCHES "^(/lib/|/lib32/|/libx32/|/lib64/|/var/empty/lib/|/var/empty/lib32/|/var/empty/libx32/|/var/empty/lib64/|/var/empty/X11R6/|/var/empty/bin/)")
+      if(resolved_file MATCHES "^(/lib/|/lib32/|/libx32/|/lib64/|/usr/lib/|/usr/lib32/|/usr/libx32/|/usr/lib64/|/usr/X11R6/|/usr/bin/)")
         set(is_system 1)
       endif()
     endif()
 
     if(APPLE)
-      if(resolved_file MATCHES "^(/System/Library/|/var/empty/lib/)")
+      if(resolved_file MATCHES "^(/System/Library/|/usr/lib/)")
         set(is_system 1)
       endif()
     endif()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackDeb.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackDeb.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackDeb.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackDeb.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -232,7 +232,7 @@
       endif()
 
       # Split out the debug symbols from the binaries
-      set(_FILE_DBGSYM ${_DBGSYM_ROOT}/var/empty/lib/debug/.build-id/${_BUILD_ID_START}/${_BUILD_ID_REMAINING}.debug)
+      set(_FILE_DBGSYM ${_DBGSYM_ROOT}/usr/lib/debug/.build-id/${_BUILD_ID_START}/${_BUILD_ID_REMAINING}.debug)
       get_filename_component(_OUT_DIR "${_FILE_DBGSYM}" DIRECTORY)
       file(MAKE_DIRECTORY "${CPACK_TEMPORARY_DIRECTORY}/${_OUT_DIR}")
       execute_process(COMMAND ${OBJCOPY_EXECUTABLE} --only-keep-debug "${_FILE}" "${_FILE_DBGSYM}"
@@ -652,7 +652,7 @@
   foreach(_FILE IN LISTS CPACK_DEB_SHARED_OBJECT_FILES)
     get_filename_component(_DIR ${_FILE} DIRECTORY)
     # all files in CPACK_DEB_SHARED_OBJECT_FILES have dot at the beginning
-    if(_DIR STREQUAL "./lib" OR _DIR STREQUAL "./var/empty/lib")
+    if(_DIR STREQUAL "./lib" OR _DIR STREQUAL "./usr/lib")
       set(CPACK_ADD_LDCONFIG_CALL 1)
     endif()
   endforeach()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackRPM.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackRPM.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackRPM.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Internal/CPack/CPackRPM.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -184,7 +184,7 @@
   endif()
 
   if(NOT DEFINED CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST)
-    set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST /etc /etc/init.d /var/empty /var/empty/bin
+    set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST /etc /etc/init.d /var/empty /usr/bin
         /var/empty/include /var/empty/lib /var/empty/libx32 /var/empty/lib64
         /var/empty/share /var/empty/share/aclocal /var/empty/share/doc )
     if(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION)
@@ -696,7 +696,7 @@
   endif()
 
   if(NOT DEFINED CPACK_RPM_DEBUGINFO_EXCLUDE_DIRS)
-    set(CPACK_RPM_DEBUGINFO_EXCLUDE_DIRS /var/empty /var/empty/src /var/empty/src/debug)
+    set(CPACK_RPM_DEBUGINFO_EXCLUDE_DIRS /var/empty /usr/src /var/empty/src/debug)
     if(CPACK_RPM_DEBUGINFO_EXCLUDE_DIRS_ADDITION)
       if(CPACK_RPM_PACKAGE_DEBUG)
         message("CPackRPM:Debug: Adding ${CPACK_RPM_DEBUGINFO_EXCLUDE_DIRS_ADDITION} to builtin omit list.")
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/AIX.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/AIX.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/AIX.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/AIX.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -7,7 +7,7 @@
 # RPATH support on AIX is called libpath.  By default the runtime
 # libpath is paths specified by -L followed by /var/empty/lib and /lib.  In
 # order to prevent the -L paths from being used we must force use of
-# -Wl,-blibpath:/var/empty/lib:/lib whether RPATH support is on or not.
+# -Wl,-blibpath:/usr/lib:/lib whether RPATH support is on or not.
 # When our own RPATH is to be added it may be inserted before the
 # "always" paths.
 if(NOT DEFINED CMAKE_PLATFORM_REQUIRED_RUNTIME_PATH)
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Common.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Common.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Common.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Common.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -207,11 +207,11 @@
     endif()
     if(NOT CMAKE_ANDROID_NDK_DEPRECATED_HEADERS)
       list(APPEND CMAKE_${lang}_STANDARD_INCLUDE_DIRECTORIES
-        "${_cmake_sysroot_compile}/var/empty/include"
-        "${_cmake_sysroot_compile}/var/empty/include/${CMAKE_ANDROID_ARCH_TRIPLE}"
+        "${_cmake_sysroot_compile}/usr/include"
+        "${_cmake_sysroot_compile}/usr/include/${CMAKE_ANDROID_ARCH_TRIPLE}"
         )
     else()
-      list(APPEND CMAKE_${lang}_STANDARD_INCLUDE_DIRECTORIES "${_cmake_sysroot_compile}/var/empty/include")
+      list(APPEND CMAKE_${lang}_STANDARD_INCLUDE_DIRECTORIES "${_cmake_sysroot_compile}/usr/include")
     endif()
     unset(_cmake_sysroot_compile)
   endif()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Determine.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Determine.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Determine.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Android-Determine.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -136,12 +136,12 @@
       "does not exist."
       )
   endif()
-  if(NOT EXISTS "${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/var/empty/include/android/api-level.h")
+  if(NOT EXISTS "${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/usr/include/android/api-level.h")
     message(FATAL_ERROR
       "Android: The standalone toolchain directory specified by CMAKE_ANDROID_STANDALONE_TOOLCHAIN:\n"
       "  ${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}\n"
       "does not contain a sysroot with a known layout.  The file:\n"
-      "  ${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/var/empty/include/android/api-level.h\n"
+      "  ${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/usr/include/android/api-level.h\n"
       "does not exist."
       )
   endif()
@@ -181,7 +181,7 @@
     # The compiler launcher does not know __ANDROID_API__.  Assume this
     # is not unified headers and look for it in the api-level.h header.
     set(_ANDROID_API_LEVEL_H_REGEX "^[\t ]*#[\t ]*define[\t ]+__ANDROID_API__[\t ]+([0-9]+)")
-    file(STRINGS "${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/var/empty/include/android/api-level.h"
+    file(STRINGS "${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/usr/include/android/api-level.h"
       _ANDROID_API_LEVEL_H_CONTENT REGEX "${_ANDROID_API_LEVEL_H_REGEX}")
     if(_ANDROID_API_LEVEL_H_CONTENT MATCHES "${_ANDROID_API_LEVEL_H_REGEX}")
       set(_ANDROID_STANDALONE_TOOLCHAIN_API "${CMAKE_MATCH_1}")
@@ -192,7 +192,7 @@
       "Android: Did not detect API level from\n"
       "  ${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/bin/clang\n"
       "or\n"
-      "  ${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/var/empty/include/android/api-level.h\n"
+      "  ${CMAKE_ANDROID_STANDALONE_TOOLCHAIN}/sysroot/usr/include/android/api-level.h\n"
       )
   endif()
 endif()
@@ -471,7 +471,7 @@
 endif()
 
 if(CMAKE_ANDROID_NDK AND NOT DEFINED CMAKE_ANDROID_NDK_DEPRECATED_HEADERS)
-  if(IS_DIRECTORY "${CMAKE_ANDROID_NDK}/sysroot/var/empty/include/${CMAKE_ANDROID_ARCH_TRIPLE}")
+  if(IS_DIRECTORY "${CMAKE_ANDROID_NDK}/sysroot/usr/include/${CMAKE_ANDROID_ARCH_TRIPLE}")
     # Unified headers exist so we use them by default.
     set(CMAKE_ANDROID_NDK_DEPRECATED_HEADERS 0)
   else()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/CrayLinuxEnvironment.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/CrayLinuxEnvironment.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/CrayLinuxEnvironment.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/CrayLinuxEnvironment.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -7,12 +7,12 @@
   set(CMAKE_SYSTEM_VERSION "$ENV{CRAYOS_VERSION}")
 elseif(DEFINED ENV{XTOS_VERSION})
   set(CMAKE_SYSTEM_VERSION "$ENV{XTOS_VERSION}")
-elseif(EXISTS /etc/var/empty/cray/release/cle-release)
-  file(STRINGS /etc/var/empty/cray/release/cle-release release REGEX "^RELEASE=.*")
+elseif(EXISTS /etc/opt/cray/release/cle-release)
+  file(STRINGS /etc/opt/cray/release/cle-release release REGEX "^RELEASE=.*")
   string(REGEX REPLACE "^RELEASE=(.*)$" "\\1" CMAKE_SYSTEM_VERSION "${release}")
   unset(release)
-elseif(EXISTS /etc/var/empty/cray/release/clerelease)
-  file(READ /etc/var/empty/cray/release/clerelease CMAKE_SYSTEM_VERSION)
+elseif(EXISTS /etc/opt/cray/release/clerelease)
+  file(READ /etc/opt/cray/release/clerelease CMAKE_SYSTEM_VERSION)
 endif()
 
 # Guard against multiple messages
@@ -51,7 +51,7 @@
 # search types.
 list(APPEND CMAKE_SYSTEM_PREFIX_PATH
   # Standard
-  $ENV{SYSROOT_DIR}/var/empty/local $ENV{SYSROOT_DIR}/var/empty $ENV{SYSROOT_DIR}/
+  $ENV{SYSROOT_DIR}/usr/local $ENV{SYSROOT_DIR}/usr $ENV{SYSROOT_DIR}/
 
   # CMake install location
   "${__cmake_install_dir}"
@@ -70,16 +70,16 @@
 endif()
 
 list(APPEND CMAKE_SYSTEM_INCLUDE_PATH
-  $ENV{SYSROOT_DIR}/var/empty/include/X11
+  $ENV{SYSROOT_DIR}/usr/include/X11
 )
 list(APPEND CMAKE_SYSTEM_LIBRARY_PATH
-  $ENV{SYSROOT_DIR}/var/empty/local/lib64
-  $ENV{SYSROOT_DIR}/var/empty/lib64
+  $ENV{SYSROOT_DIR}/usr/local/lib64
+  $ENV{SYSROOT_DIR}/usr/lib64
   $ENV{SYSROOT_DIR}/lib64
 )
 list(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  $ENV{SYSROOT_DIR}/var/empty/local/lib64
-  $ENV{SYSROOT_DIR}/var/empty/lib64
+  $ENV{SYSROOT_DIR}/usr/local/lib64
+  $ENV{SYSROOT_DIR}/usr/lib64
   $ENV{SYSROOT_DIR}/lib64
 )
 
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin-Initialize.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin-Initialize.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin-Initialize.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin-Initialize.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -147,7 +147,7 @@
 
 function(_apple_resolve_supported_archs_for_sdk_from_system_lib sdk_path ret ret_failed)
   # Detect the supported SDK architectures by inspecting the main libSystem library.
-  set(common_lib_prefix "${sdk_path}/var/empty/lib/libSystem")
+  set(common_lib_prefix "${sdk_path}/usr/lib/libSystem")
   set(system_lib_dylib_path "${common_lib_prefix}.dylib")
   set(system_lib_tbd_path "${common_lib_prefix}.tbd")
 
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/Darwin.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -158,11 +158,11 @@
     endif()
   endforeach()
 
-  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/lib)
-    list(INSERT CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES 0 ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/lib)
+  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/usr/lib)
+    list(INSERT CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES 0 ${_CMAKE_OSX_SYSROOT_PATH}/usr/lib)
   endif()
-  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/local/lib)
-    list(INSERT CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES 0 ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/local/lib)
+  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/usr/local/lib)
+    list(INSERT CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES 0 ${_CMAKE_OSX_SYSROOT_PATH}/usr/local/lib)
   endif()
 endif()
 if (OSX_DEVELOPER_ROOT AND EXISTS "${OSX_DEVELOPER_ROOT}/Library/Frameworks")
@@ -224,16 +224,16 @@
 endif()
 
 if(_CMAKE_OSX_SYSROOT_PATH)
-  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/include)
+  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/usr/include)
     list(INSERT CMAKE_SYSTEM_PREFIX_PATH 0 ${_CMAKE_OSX_SYSROOT_PATH}/usr)
     foreach(lang C CXX OBJC OBJCXX Swift)
-      list(APPEND _CMAKE_${lang}_IMPLICIT_INCLUDE_DIRECTORIES_INIT ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/include)
+      list(APPEND _CMAKE_${lang}_IMPLICIT_INCLUDE_DIRECTORIES_INIT ${_CMAKE_OSX_SYSROOT_PATH}/usr/include)
     endforeach()
   endif()
-  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/local/include)
-    list(INSERT CMAKE_SYSTEM_PREFIX_PATH 0 ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/local)
+  if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/usr/local/include)
+    list(INSERT CMAKE_SYSTEM_PREFIX_PATH 0 ${_CMAKE_OSX_SYSROOT_PATH}/usr/local)
     foreach(lang C CXX OBJC OBJCXX Swift)
-      list(APPEND _CMAKE_${lang}_IMPLICIT_INCLUDE_DIRECTORIES_INIT ${_CMAKE_OSX_SYSROOT_PATH}/var/empty/local/include)
+      list(APPEND _CMAKE_${lang}_IMPLICIT_INCLUDE_DIRECTORIES_INIT ${_CMAKE_OSX_SYSROOT_PATH}/usr/local/include)
     endforeach()
   endif()
 endif()
diff -aru /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/HP-UX-HP.cmake /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/HP-UX-HP.cmake
--- /nix/store/ahbpir8p78zddf2fn3fg9l127rhimgqy-cmake-3.19.7/share/cmake-3.19/Modules/Platform/HP-UX-HP.cmake	1969-12-31 19:00:01.000000000 -0500
+++ /nix/store/hp22pmzm6m1rxw35a078yrh2s3gccw62-cmake-3.19.7/share/cmake-3.19/Modules/Platform/HP-UX-HP.cmake	1969-12-31 19:00:01.000000000 -0500
@@ -11,7 +11,7 @@
 macro(__hpux_compiler_hp lang)
   set(CMAKE_${lang}_COMPILE_OPTIONS_PIC "+Z")
   set(CMAKE_SHARED_LIBRARY_${lang}_FLAGS "+Z")
-  set(CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-Wl,-E,+nodefaultrpath -b -L/var/empty/lib")
+  set(CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-Wl,-E,+nodefaultrpath -b -L/usr/lib")
   set(CMAKE_SHARED_LIBRARY_LINK_${lang}_FLAGS "-Wl,-E")
   set(CMAKE_SHARED_LIBRARY_RUNTIME_${lang}_FLAG "-Wl,+b")
   set(CMAKE_SHARED_LIBRARY_RUNTIME_${lang}_FLAG_SEP ":")
```

Reply:
This should reduce the scope of changes made by fixCmakeFiles. I agree with you, @drodriguez, some things better be handled by patches.

Reply:
That would apply to a lot more cases. It does remove the problems that I am more affected by, and I am sure it fixes some problems that people might find (for example the changes to the Android file). However, I think it has some false positives that I am not sure if they will affect other Nix packages. Some of the cases are when `/usr` is used attached to a switch like `-L`, `-rpath,`, and others. There's other examples in the above diff that should be evaluated (there's usage of `/usr` in the middle of regexes, which means that things like `^` and `|` might need to be considered.

To be clear, if you want me to change the PR to use your regex, I am not opposing. I will double check, but I think it fixes my problem as well as my version, but I think it introduces a lot more changes and might affect a lot more packages.

Also, if you prefer it, I can leave the `sed` line as it was before, and post-process the files I am interested in fixing with a simple `substituteInPlace` (sadly `patches` is applied just before `preConfigure`).

Reply:
Looking at the diff, I also spotted the things you've highlighted, but did not spot anything outstandingly wrong with the changes. I don't think that fixCmakeFiles is intended as some sort of robust solution for all cases. I would certainly not expect it to fix our regexes logic and HP-UX support at the same time.